src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -610,6 +583,19 @@ putIndex arr i x =
     unsafeWithArrayContents (arrContents arr) (arrStart arr)
         $ \p -> putIndexPtr p (aEnd arr) i x
 
+-------------------------------------------------------------------------------
+-- Random writes
+-------------------------------------------------------------------------------
```

Comment:
This header should remain at its original place. It is for all the random write functions.